src/Adapter/Debug/DebugProfiling.php

```diff
@@ -0,0 +1,187 @@
+    public const DEBUG_PROFILING_ERROR_NO_DEFINITION_FOUND = 5;
+
+    /**
+     * Is the profiler enabled? Checks on custom defines file first.
+     *
+     * @return bool Whether debug profiling is enabled
+     */
+    public function isProfilingEnabled(): bool
+    {
+        $definesClean = '';
+        $customDefinesPath = _PS_ROOT_DIR_ . '/config/defines_custom.inc.php';
+        $definesPath = _PS_ROOT_DIR_ . '/config/defines.inc.php';
+
+        if (is_readable($customDefinesPath)) {
+            $definesClean = php_strip_whitespace($customDefinesPath);
+        }
+
+        if (!preg_match('/define\(\'_PS_DEBUG_PROFILING_\', ([a-zA-Z]+)\);/Ui', $definesClean, $debugProfilingValue)) {
+            $definesClean = php_strip_whitespace($definesPath);
+            if (!preg_match('/define\(\'_PS_DEBUG_PROFILING_\', ([a-zA-Z]+)\);/Ui', $definesClean, $debugProfilingValue)) {
+                return false;
+            }
+        }
+
+        return 'true' === Tools::strtolower($debugProfilingValue[1]);
+    }
+
+    /**
+     * Enable Debug profiling.
+     *
+     * @return int Whether changing debug profiling succeeded or error code
+     */
+    public function enable(): int
+    {
+        return $this->changeProfilingValue('true');
+    }
+
+    /**
+     * Disable debug profiling.
+     *
+     * @return int Whether changing debug profiling succeeded or error code
+     */
+    public function disable(): int
+    {
+        return $this->changeProfilingValue('false');
+    }
+
+    /**
+     * Check read permission on custom defines.inc.php.
+     *
+     * @return bool Whether the file can be read
+     */
+    private function isCustomDefinesReadable(): bool
+    {
+        return is_readable(_PS_ROOT_DIR_ . '/config/defines_custom.inc.php');
+    }
+
+    /**
+     * Check read permission on main defines.inc.php.
+     *
+     * @return bool Whether the file can be read
+     */
+    private function isMainDefinesReadable(): bool
+    {
+        return is_readable(_PS_ROOT_DIR_ . '/config/defines.inc.php');
+    }
+
+    /**
+     * Update Debug profiling value in main defines file.
+     *
+     * @param string $value should be "true" or "false"
+     *
+     * @return int the debug profiling
+     */
+    private function updateProfilingValueInMainFile(string $value): int
+    {
+        $filename = _PS_ROOT_DIR_ . '/config/defines.inc.php';
+        $cleanedFileContent = php_strip_whitespace($filename);
+        $fileContent = Tools::file_get_contents($filename);
+
+        if (!preg_match('/define\(\'_PS_DEBUG_PROFILING_\', ([a-zA-Z]+)\);/Ui', $cleanedFileContent)) {
+            return self::DEBUG_PROFILING_ERROR_NO_DEFINITION_FOUND;
+        }
+
+        $fileContent = preg_replace('/define\(\'_PS_DEBUG_PROFILING_\', ([a-zA-Z]+)\);/Ui', 'define(\'_PS_DEBUG_PROFILING_\', ' . $value . ');', $fileContent);
+        if (!@file_put_contents($filename, $fileContent)) {
```

Comment:
🤯 there is no PHP function to simply enable or disable profiling? it needs a file to be written?

this is crazy 😄 @Hlavtox I'm sorry I'm not against your PR it's a lot of work but ... I think there must be a simpler solution. right now to turn on a setting we are _writing on the disk_ a new file this is over-engineered

can we imagine a solution where we create a new SQL table called 'dynamic defines' and in this SQL table we put key/row values, and in the `/config/defines.inc.php` at the very beginning we load every defines from the DB (and put it in cache)?

then enabling or disabling any setting becomes a simple query, not a complex I/O operation

Reply:
Nope, the profiling settings is hardcoded in files the config folder. This is an old-school legacy since first days of PrestaShop. 😄

I just copied the same logic that controls the DebugMode. 👍 
https://github.com/PrestaShop/PrestaShop/blob/develop/src/Adapter/Debug/DebugMode.php

We can make an unified class that will handle modifying the config file later, but in order not to introduce any BC breaks and increasing the scope, I kept it like this.

Reply:
> can we imagine a solution where we create a new SQL table called 'dynamic defines' and in this SQL table we put key/row values, and in the /config/defines.inc.php at the very beginning we load every defines from the DB (and put it in cache)?

Also, some settings are there on purpose:
- Profiling - How you are gonna measure from the start, if you don't know if you should, because the communication with database starts later?
- Debug mode - You need error reporting sooner than the Db communication starts. How do you diagnose error when connecting to db?
- There are also security concerns. The recent attack on Prestashop was because the attacker was able to change smarty cache settings in the database. If it was in a file, he would not be able to.

Reply:
@matks, as Daniel said, it's the same as the developer mode, I think we can accept it as is, and eventually, in the future, this has to be configured with .env, right? :)

Reply:
All right 😄 I don't want to block this wonderful block so OK for me

But we need to think about something better for future ;)

Reply:
> But we need to think about something better for future ;)
Definitely!